.circleci/config.yml

```diff
@@ -34,6 +34,20 @@ jobs:
             pip install .
             pip install scipy
 
+      - run:
+          name: create release notes
+          command: |
+            . venv/bin/activate
+            pip install git+https://github.com/hawkowl/towncrier.git@master
+            VERSION=$(python -c "import setup; print(setup.VERSION)")
+            towncrier --version $VERSION --yes
+            NOT_REMOVED=$(find doc/release/upcoming_changes -mindepth 1 -not -name "README.rst" -not -name "template.rst")
+            if [ -n "$NOT_REMOVED" ]; then
+                echo "Error, the following news fragments were not used:"
+                echo $NOT_REMOVED
+                false
+            fi
+
```

Comment:
If you are already adding a test...

This hardcodes the path. What if you add a `tool/check_towncrier.py` to do this in python (where everything would be easier including reading the toml file), or use some advanced grep/sed/tr like this
```
grep "\[tool.towncrier\]" -A10 pyproject.toml | grep "directory.*=" | sed -e"s/^.*=//" |tr -d ' "'
```
to get the directory name? Otherwise this check may silently fail (it does print to stderr but that is not checked) if we move the directory

Reply:
Mostly did it because I noticed there was already one wrong (although probably mainly due renames). OK, I will go with your grep for now.

Reply:
nvm. Writing a python script is much nicer...